docs/development.md

```diff
@@ -1,5 +1,42 @@
 # Development
 
+## golangci-lint
+
+A [golangci-lint](https://golangci-lint.run/) [config](../.golangci.yaml) and [Makefile](../Makefile) target are provided to keep the codebase aligned to certain best practices and styles.
+
+The target installs golangci-lint if it is not already present. To run use the following command:
+
+```shell
+make lint
+test -s /src/ssp-operator/bin/golangci-lint || curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b /src/ssp-operator/bin v1.51.1
+/src/ssp-operator/bin/golangci-lint run --timeout 5m
+```
+
+The target is also used when running unit tests locally.
+
+## pre-commit
+
+A [pre-commit](https://pre-commit.com/) [config](../.pre-commit-config.yaml) is provided to help developers catch any simple mistakes prior to committing their changes.
+
+To install and use the tool please run the following commands:
+
+```shell
+pip install --user pre-commit
+pre-commit install
+git commit -s
+[..]
+trim trailing whitespace.................................................Passed
+fix end of files.........................................................Passed
+check yaml...........................................(no files to check)Skipped
+check for added large files..............................................Passed
+check for merge conflicts................................................Passed
+check that executables have shebangs.................(no files to check)Skipped
+check that scripts with shebangs are executable..........................Passed
+detect private key.......................................................Passed
+mixed line ending........................................................Passed
+lint.....................................................................Passed
```

Comment:
The same here. Can you keep only input? Or show the output in a different block?